internal/efi/default_env.go

```diff
@@ -141,41 +199,30 @@ func (d *defaultEnvSysfsDevice) AttributeReader(attr string) (rc io.ReadCloser,
 	return f, nil
 }
 
-// DeviceForClass implements [HostEnvironment.DevicesForClass].
-func (defaultEnvImpl) DevicesForClass(class string) ([]SysfsDevice, error) {
-	classPath := filepath.Join(sysfsPath, "class", class)
-	f, err := os.Open(classPath)
-	switch {
-	case os.IsNotExist(err):
-		// it's ok to have no devices for the specified class
-		return nil, nil
-	case err != nil:
-		return nil, err
-	}
-	defer f.Close()
-
-	entries, err := f.ReadDir(-1)
-	if err != nil {
-		return nil, err
-	}
-
-	var out []SysfsDevice
-	for _, entry := range entries {
-		path, err := filepath.EvalSymlinks(filepath.Join(classPath, entry.Name()))
-		if err != nil {
-			return nil, fmt.Errorf("cannot resolve path for %s: %w", entry.Name(), err)
-		}
-		subsystem, err := filepath.EvalSymlinks(filepath.Join(path, "subsystem"))
-		if err != nil {
-			return nil, fmt.Errorf("cannot resolve subsystem for %s: %w", entry.Name(), err)
+// EnumerateDevices implements [HostEnvironment.EnumerateDevices].
+func (defaultEnvImpl) EnumerateDevices(matcher netlink.Matcher) ([]SysfsDevice, error) {
+	queue := make(chan crawler.Device)
+	errs := make(chan error)
+	crawlerExistingDevices(queue, errs, matcher)
+
+	var devices []SysfsDevice
+
+	for {
+		select {
+		case dev, more := <-queue:
+			if !more {
+				return devices, nil
+			}
+			devices = append(devices, &defaultEnvSysfsDevice{
+				path:      dev.KObj,
+				props:     dev.Env,
+				subsystem: dev.Env["SUBSYSTEM"],
+			})
+			delete(dev.Env, "SUBSYSTEM")
```

Comment:
can you comment on the why of this delete?

Reply:
I've added a comment for this now.